citesphere/src/main/java/edu/asu/diging/citesphere/core/service/impl/AuthorityService.java

```diff
@@ -108,6 +132,23 @@ public IAuthorityEntry find(String id) {
         return entries;
     }
     
+    @Override
+    public Set<IAuthorityEntry> findByNameInDataset(String name, String citationGroupId) throws GroupDoesNotExistException {
+        Optional<CitationGroup> group = groupRepository.findById(new Long(citationGroupId));
+        if (!group.isPresent()) {
+            throw new GroupDoesNotExistException("Group with id " + citationGroupId + " does not exist.");
+        }
+        List<Person> persons = personAuthorityRepository.findPersonsByCitationGroupAndNameLike((ICitationGroup)group.get(), name);
```

Comment:
this should be paginated. You don't want to iterate over 100s of entries (worst case)